File: src/neetcode/Java/_7ReverseInteger.java
package neetcode.Java;

public class _7ReverseInteger {

    public static int reverse(int x) {
        System.out.println(x);
        long ans = solving(x);
        System.out.println(ans);
        if (ans > Integer.MAX_VALUE || ans < Integer.MIN_VALUE) return 0;
        return (int) ans;
    }

    private static long solving(int num) {
        long ans = 0;
        while (num != 0) {
            int rem = num % 10;
            ans = (ans * 10) + rem;
            num /= 10;
        }
        return ans;
    }

    public static int reverse1(int x) {
        boolean isNeg;
        isNeg = x >= 0 ? false : true;
        double xn = x;
        xn = isNeg ? 0.0 - xn : xn;
        String s = xn + "";
        String ss = "";
        for (int i = 0; i < s.length(); i++) {
            ss = s.charAt(i) + ss;
        }
        try {
            xn = Double.parseDouble(ss);
        } catch (Exception e) {
            return 0;
        }
        xn = isNeg ? -xn : xn;
        if (xn >= Integer.MIN_VALUE && xn < Integer.MAX_VALUE) {
            return (int) xn;
        }
        return 0;
    }

    public static void main(String[] args) {
        int res = reverse(-2147483648);
        System.out.println(res);
    }
}

File: src/neetcode/kotlin/_7ReverseInteger.kt
package neetcode.kotlin

object _7ReverseInteger {
    fun reverse(x: Int): Int {
        println(x)
        val ans = solving(x)
        println(ans)
        if (ans > Int.MAX_VALUE || ans < Int.MIN_VALUE) return 0
        return ans.toInt()
    }

    private fun solving(num: Int): Long {
        var num = num
        var ans: Long = 0
        while (num != 0) {
            val rem = num % 10
            ans = (ans * 10) + rem
            num /= 10
        }
        return ans
    }

    fun reverse1(x: Int): Int {
        val isNeg = if (x >= 0) false else true
        var xn = x.toDouble()
        xn = if (isNeg) 0.0 - xn else xn
        val s = xn.toString() + ""
        var ss = ""
        for (i in 0 until s.length) {
            ss = s[i].toString() + ss
        }
        try {
            xn = ss.toDouble()
        } catch (e: Exception) {
            return 0
        }
        xn = if (isNeg) -xn else xn
        if (xn >= Int.MIN_VALUE && xn < Int.MAX_VALUE) {
            return xn.toInt()
        }
        return 0
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val res = reverse(-2147483648)
        println(res)
    }
}
